internal/transport/http2_server.go

```diff
@@ -1268,6 +1278,9 @@ func (t *http2Server) Close(err error) {
 	channelz.RemoveEntry(t.channelz.ID)
 	// Cancel all active streams.
 	for _, s := range streams {
+		if s.deadlineTimerCancel != nil {
+			s.deadlineTimerCancel()
+		}
```

Comment:
Should we make sure this is always a valid function so we can just call it without the nil check?